### PR TITLE
Cleaned up some routes

### PR DESCRIPTION
**Pull request by wyattjoh**
## What does this PR do?

- Cleans up some routes to standardize

### CODE REVIEW

File: routes/api/account/index.js [hunk 1]
```diff
@@ -121,12 +119,8 @@ router.put('/username', authorization.needed(), (req, res, next) => {
...
       res.status(204).end();
     })
-    .catch(error => {
-      if (error.code === 11000) {
-        next(errors.ErrUsernameTaken);
-      } else {
-        next(error);
-      }
+    .catch((err) => {
+      next(err);
```
Comment (gabelula) [inline routes/api/account/index.js:123]: Where are we managing the type of error now? Does this work to return the right error in FE?

File: routes/api/assets/index.js [hunk 2]
```diff
@@ -79,7 +79,8 @@ router.get('/:asset_id', (req, res, next) => {
     .findById(req.params.asset_id)
     .then((asset) => {
       if (!asset) {
-        return res.status(404).end();
+        res.status(404).end();
+        return;
```
Comment (gabelula) [inline routes/api/assets/index.js:83]: Why this change?

File: routes/api/account/index.js (hunk 1 above)
Comment (gabelula) [reply routes/api/account/index.js:123]: I see that it is resolved in the user service.

File: routes/api/assets/index.js (hunk 2 above)
Comment (wyattjoh) [reply routes/api/assets/index.js:83]: It wasn't documented what `res.status(404).end();`, so I didn't feel it necessary for it to return it's result through the promise chain.

**Review by gabelula — APPROVED** [review]
Fine if nobody is using the results of res.status(404).end() from that endpoint.
